### PR TITLE
Cancel drawing operation with Esc

### PR DESCRIPTION
**Pull request by drucelweisse**
# Description

This pr add handling esc button pressed on dragging action

## Changes

- added cancel() to DraggingAction and implementations for existing actions
- added EscModifierActionDelegate to trigger esc pressed action
- added EscModifierAction protocol to handle keyboard events
- added EscModifierActionCommand with keyListener
- added EmptyEscModifierAction

## TODO:

- [x] tests for canceling actions

## Screenshots

https://user-images.githubusercontent.com/36012972/185763102-974cfa53-75c1-4afe-9b1b-c28b53544b93.mov

### CODE REVIEW

**Comment by drucelweisse** [comment]
Also I don't like that key listeners from NSEvent return as Any?, should we make custom publisher for that?

**Review by akaDuality — APPROVED** [review]
Cool tests

File: Document/Sources/Document/Drawing/EscCancelAction/EscModifierAction.swift [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+        return EscModifierActionCommand()
+        #else
+        return EmptyEscModifierAction()
+        #endif
+    }
+}
+
+
+public protocol EscModifierActionDelegate: AnyObject {
+    func didPressed()
```
Comment (akaDuality) [inline Document/Sources/Document/Drawing/EscCancelAction/EscModifierAction.swift:26]: `didPressedEcs` would be better

File: VoiceOver Designer/Features/Tests/EditorTests/CancellingActionsTests.swift [hunk 2]
```diff
@@ -0,0 +1,90 @@
...
+            controller.controlsView.copyListener = copyCommand
+            drawRect_10_60()
+        }
+        
+        // Copy
+        
+        await MainActor.run {
+            copyCommand.isCopyHold = true
+            sut.mouseDown(on: .coord(15))
+            controller.controlsView.escListener.delegate?.didPressed()
```
Comment (akaDuality) [inline VoiceOver Designer/Features/Tests/EditorTests/CancellingActionsTests.swift:27]: Better to wrap this line to some domain specific function like `pressEsc()` and hide implementation details

**Pull request merged**